subsys/pm/policy/policy_residency.c

```diff
@@ -4,48 +4,72 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <devicetree.h>
 #include <zephyr.h>
 #include <kernel.h>
 #include <pm/pm.h>
 #include <pm/policy.h>
+#include <sys/check.h>
 
 #define LOG_LEVEL CONFIG_PM_LOG_LEVEL /* From power module Kconfig */
 #include <logging/log.h>
 LOG_MODULE_DECLARE(power);
 
-static const struct pm_state_info pm_min_residency[] =
-	PM_STATE_INFO_DT_ITEMS_LIST(DT_NODELABEL(cpu0));
+#if DT_NODE_EXISTS(DT_PATH(cpus))
+
+#define CPU_STATES(n) (struct pm_state_info[])PM_STATE_INFO_DT_ITEMS_LIST(n),
+
+static const struct pm_state_info *pm_min_residency[] = {
+	DT_FOREACH_CHILD(DT_PATH(cpus), CPU_STATES)
+};
+
+#define CPU_STATES_SIZE(n) PM_STATE_DT_ITEMS_LEN(n),
+
+static int pm_min_residency_sizes[] = {
+	DT_FOREACH_CHILD(DT_PATH(cpus), CPU_STATES_SIZE)
+};
+#else
+static const struct pm_state_info *pm_min_residency[CONFIG_MP_NUM_CPUS] = {};
+static int pm_min_residency_sizes[CONFIG_MP_NUM_CPUS] = {};
```

Comment:
BUILD_ASSERT added to enforce cpus node.

Reply:
I reverted the change. If I add the BUILD_ASSERT, all native_posix because it does not implement it. For now I think it is fine to keep the old behavior. I don't want to just put `HAS_NO_PM` for native posix because many pm tests are using native posix only.